sparse/unit_test/Test_Sparse_spiluk.hpp

```diff
@@ -26,486 +26,464 @@
-  hentries(6)  = 0;
-  hentries(7)  = 3;
-  hentries(8)  = 4;
-  hentries(9)  = 0;
-  hentries(10) = 4;
-  hentries(11) = 1;
-  hentries(12) = 5;
-  hentries(13) = 2;
-  hentries(14) = 6;
-  hentries(15) = 3;
-  hentries(16) = 4;
-  hentries(17) = 7;
-  hentries(18) = 3;
-  hentries(19) = 4;
-  hentries(20) = 8;
-
-  hvalues(0)  = 10;
-  hvalues(1)  = 0.3;
-  hvalues(2)  = 0.6;
-  hvalues(3)  = 11;
-  hvalues(4)  = 0.7;
-  hvalues(5)  = 12;
-  hvalues(6)  = 5;
-  hvalues(7)  = 13;
-  hvalues(8)  = 1;
-  hvalues(9)  = 4;
-  hvalues(10) = 14;
-  hvalues(11) = 3;
-  hvalues(12) = 15;
-  hvalues(13) = 7;
-  hvalues(14) = 16;
-  hvalues(15) = 6;
-  hvalues(16) = 5;
-  hvalues(17) = 17;
-  hvalues(18) = 2;
-  hvalues(19) = 2.5;
-  hvalues(20) = 18;
-
-  typename KernelHandle::const_nnz_lno_t fill_lev = 2;
-
-  for (int i = 0; i < nstreams; i++) {
-    // Allocate A as input
-    A_row_map_v[i] = RowMapType("A_row_map", nrows + 1);
-    A_entries_v[i] = EntriesType("A_entries", nnz);
-    A_values_v[i]  = ValuesType("A_values", nnz);
-
-    // Copy from host to device
-    Kokkos::deep_copy(A_row_map_v[i], hrow_map);
-    Kokkos::deep_copy(A_entries_v[i], hentries);
-    Kokkos::deep_copy(A_values_v[i], hvalues);
-
-    // Create handle
-    kh_v[i] = KernelHandle();
-    if (test_algo == 0)
-      kh_v[i].create_spiluk_handle(SPILUKAlgorithm::SEQLVLSCHD_RP, nrows,
-                                   4 * nrows, 4 * nrows);
-    else if (test_algo == 1)
-      kh_v[i].create_spiluk_handle(SPILUKAlgorithm::SEQLVLSCHD_TP1, nrows,
-                                   4 * nrows, 4 * nrows);
-    kh_ptr_v[i] = &kh_v[i];
-
-    auto spiluk_handle = kh_v[i].get_spiluk_handle();
-    std::cout << "  Stream " << i << ": ";
-    spiluk_handle->print_algorithm();
-
-    // Allocate L and U as outputs
-    L_row_map_v[i] = RowMapType("L_row_map", nrows + 1);
-    L_entries_v[i] = EntriesType("L_entries", spiluk_handle->get_nnzL());
-    L_values_v[i]  = ValuesType("L_values", spiluk_handle->get_nnzL());
-    U_row_map_v[i] = RowMapType("U_row_map", nrows + 1);
-    U_entries_v[i] = EntriesType("U_entries", spiluk_handle->get_nnzU());
-    U_values_v[i]  = ValuesType("U_values", spiluk_handle->get_nnzU());
-
-    // Symbolic phase
-    spiluk_symbolic(kh_ptr_v[i], fill_lev, A_row_map_v[i], A_entries_v[i],
-                    L_row_map_v[i], L_entries_v[i], U_row_map_v[i],
-                    U_entries_v[i], nstreams);
+    if (!run_streams_test) return;
+
+    std::vector<execution_space> instances;
+    if (nstreams == 1)
+      instances = Kokkos::Experimental::partition_space(execution_space(), 1);
+    else if (nstreams == 2)
+      instances =
+          Kokkos::Experimental::partition_space(execution_space(), 1, 1);
+    else if (nstreams == 3)
+      instances =
+          Kokkos::Experimental::partition_space(execution_space(), 1, 1, 1);
+    else
+      instances =
+          Kokkos::Experimental::partition_space(execution_space(), 1, 1, 1, 1);
```

Comment:
We can generalize the partition_space creation by this:
```
std::vector<execution_space> instances;
std::vector<int> weights(nstreams);
std::fill(weights.begin(), weights.end(), 1);
instances = Kokkos::Experimental::partition_space(execution_space(), weights);
```

Reply:
This was in the original (before this PR) code, but this looks like a good change so I made it.